src/Dotnet.Script.Core/Scaffolder.cs

```diff
@@ -1,46 +1,59 @@
-﻿using System;
+﻿using Dotnet.Script.Core.Templates;
+using Dotnet.Script.DependencyModel.Environment;
+using Newtonsoft.Json.Linq;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
-using System.Reflection;
+using System.Runtime.InteropServices;
 using System.Text.RegularExpressions;
-using Dotnet.Script.Core.Templates;
-using Dotnet.Script.DependencyModel.Environment;
-using Dotnet.Script.DependencyModel.Logging;
-using Dotnet.Script.DependencyModel.ProjectSystem;
-using Newtonsoft.Json.Linq;
 
 namespace Dotnet.Script.Core
 {
     public class Scaffolder
     {
-        private ScriptEnvironment _scriptEnvironment;        
+        private ScriptEnvironment _scriptEnvironment;
         private const string DefaultScriptFileName = "main.csx";
         private ScriptConsole _scriptConsole = ScriptConsole.Default;
 
         public Scaffolder()
         {
-            _scriptEnvironment = ScriptEnvironment.Default;            
+            _scriptEnvironment = ScriptEnvironment.Default;
         }
 
         public void InitializerFolder(string fileName, string currentWorkingDirectory)
-        {            
+        {
             CreateLaunchConfiguration(currentWorkingDirectory);
             CreateOmniSharpConfigurationFile(currentWorkingDirectory);
-            CreateScriptFile(fileName, currentWorkingDirectory);            
+            CreateScriptFile(fileName, currentWorkingDirectory);
         }
 
         public void CreateNewScriptFile(string fileName, string currentDirectory)
         {
             _scriptConsole.WriteNormal($"Creating '{fileName}'");
-            if(!Path.HasExtension(fileName))
+            if (!Path.HasExtension(fileName))
             {
                 fileName = Path.ChangeExtension(fileName, ".csx");
             }
             var pathToScriptFile = Path.Combine(currentDirectory, fileName);
             if (!File.Exists(pathToScriptFile))
             {
                 var scriptFileTemplate = TemplateLoader.ReadTemplate("helloworld.csx.template");
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ||
+                    RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    // add a shebang to set dotnet-script as the interpreter for .csx files
+                    scriptFileTemplate = "#!/usr/bin/env dotnet-script\n" + scriptFileTemplate;
+                }
+
                 File.WriteAllText(pathToScriptFile, scriptFileTemplate);
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ||
+                    RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    // mark .csx file as executable, this activates the shebang to run dotnet-script as interpreter
+                    Process.Start($"/bin/chmod", $"+x {pathToScriptFile}").WaitForExit();
```

Comment:
I think it would make sense to spit out there errors from these external processes, like https://github.com/filipw/dotnet-script/blob/de824d38292f3ca00563bb5e9299de888b4be38f/src/Dotnet.Script.DependencyModel/Process/CommandRunner.cs#L54

Reply:
can I just use the command runner?

Reply:
Yup, that is okay. If fact, better :)